File: 0102/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if root is None: return []
        temp = [root]
        
        res = list()
        while(temp):
            res.append([])
            temp_r = temp.copy()
            temp = list()
            for r in temp_r:
                res[-1].append(r.val)
                if r.left:
                    temp.append(r.left)
                if r.right:
                    temp.append(r.right)
        if not res[-1]: del res[-1]
        return res

File: 0102/main.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> r;
        TreeNode *p=root, *last;
        vector<vector<int>> nums;
        if(!root) return nums;
        
        r.push(root);
        last = root;
        while(!r.empty()) {
            last = r.back();
            vector<int> num;
            do {
                p = r.front();
                if(p->left) r.push(p->left);
                if(p->right) r.push(p->right);
                num.push_back(p->val);
                r.pop();
            } while(p!=last);
            nums.push_back(num);
        }
        return nums;
    }
};